src/OutsideClickHandler.jsx

```diff
@@ -45,8 +45,9 @@ export default class OutsideClickHandler extends React.Component {
     if (!disabled) this.addMouseDownEventListener(useCapture);
   }
 
-  componentWillReceiveProps({ disabled, useCapture }) {
-    const { disabled: prevDisabled } = this.props;
+  componentDidUpdate(prevProps) {
```

Comment:
I'm not confident this is a "fix"; this will cause this logic to apply after the update, instead of before.

Reply:
Hi Jordan, probably you're right but actually I couldn't find a logic that apply in a before update.

The new React API doesn't give any option for check future props (static getDerivedState..., but not correct for this case). Am i wrong ?

Reply:
I believe you are right, which means that there may not yet *be* a solution here.

If it turns out we can migrate to cDU, then we should have been doing that in the first place; if not, then we won't be compatible with React 17 in async mode until React adds a new API to replace what it's removing.

Reply:
Okay, we could just rename UNSAFE_componentWillMount meanwhile.
Anyways, I've tried this supposed "fix", on a website we're building and looks there's nothing broken, test passed too.

Reply:
That method is only available in React 16.3+, so that's also not an option.